### DISCUSSION TITLE
How should I perform such a rest query?

### DISCUSSION THREAD

**Discussion by plamber** [pnp/pnpcore#1528] [General] [answered]
Hi,

I’m trying to execute the following REST query using PnP Core:

```
yourSiteUrl/_api/web/lists('c8499167-df77-4c30-a0be-241ae768098c')/items?$top=1000&$select=Id,Title,FileLeafRef,HasUniqueRoleAssignments,EncodedAbsUrl,ContentTypeId,FileSystemObjectType,ParentList/Id,ParentList/Title,ParentList/ParentWebUrl&$expand=ParentList
```

My goal is to perform a REST query and retrieve all items from the list.

So far, I’ve successfully queried using this code:
```
await myList.Items.LoadAsync(p => p.Id, p => p.Title, p => p.HasUniqueRoleAssignments, p => p.ParentList);
```
However, I’m unsure how to include additional properties such as FileLeafRef or EncodedAbsUrl in the query and how to specify the top parameter.

Any guidance would be greatly appreciated!

Thank you,
Patrick

**Comment by jansenbe** [comment, marked as answer]
See https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#reading-list-items for the options.
